include/Dialect/Secret/IR/SecretOps.td

```diff
@@ -154,5 +154,35 @@ def Secret_GenericOp : Secret_Op<"generic", [
   let hasVerifier = 1;
 }
 
+def CastOp : Secret_Op<"cast", [Pure]> {
+  let summary = "A placeholder cast from one secret type to another";
+  let description = [{
+    A `cast` operation represents a type cast from one secret type to another,
+    that is used to enable the intermixing of various equivalent secret types
+    before a lower-level FHE scheme has been chosen.
+
+    For example, `secret.cast` can be used to convert a `secret<i8>` to a
+    `secret<tensor<8xi1>>` as a compatibility layer between boolean and
+    non-boolean parts of a program. The pass that later lowers the IR to
+    specific FHE schemes would need to replace these casts with appropriate
+    scheme-specific operations, and it is left to those later passes to
+    determine which casts are considered valid.
```

Comment:
Hmm.. I think I agree that no verifiers should check that there's equality in the # of bits / no loss of information. I think this could end up representing information loss or casting to a larger type, depending on user constraints for how much leeway there is anyway (maybe we cast to some representation with less precision)